### DISCUSSION TITLE
NearSt Adoption of Laconia.js

### DISCUSSION THREAD

**Discussion by ceilfors** [nearst/laconia#913] [General]
Hello @laconiajs/maintainers,

**TL;DR** _Your thoughts and insights on this matter are invaluable. Please review and respond to the proposal below in two weeks. Share any concerns, suggestions, or questions you may have._

Laconia.js is currently having a dependency on AWS SDK v2. Unfortunately, [AWS will stop](https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html) node16 AWS Lambda updates on 15 Aug 2024. Projects that are using Laconia.js will find it difficult to upgrade to node18 in AWS Lambda, given that the upgrade implies an upgrade to AWS SDK v3, which is incompatible with Laconia.js.

You might have seen that Laconia.js had no active development recently. I built this framework in my own free time after numerous pains in building 4 serverless apps. Since then, I have moved to projects with a different setup, and most of my free time is spent parenting. I find it now too challenging to maintain Laconia.js.

@tschoffelen from NearSt, has kindly reached out to me recently in proposing to adopt Laconia.js. Their primary driver is to become a player in the open-source space. They're one of the users of Laconia.js I know, and I've been working with the team for over 2 years.

My primary goal is for Laconia.js to continue helping its current users, and I see a great benefit in this adoption. Here are the details of the adoption proposal:
 
1. The repositories will be  transferred from `laconiajs/laconia` to `nearst/laconia`
1. The NearSt development team will maintain the repositories
    - NearSt will take responsibility for maintaining the repo, evaluating and responding to PRs
    - NearSt will host the testing infrastructure and documentation site
    - NearSt will own publish access to NPM
1. Package name remains the same `@laconia/core`.
1. License remains the same as Apache License 2.0.
1. NearSt may update the GitHub doc or documentation to show their brand
    - The relationship in between NearSt and the project is "supported/maintained by".
    - The logo will unlikely be changed to include the NearSt brand
1. As the original creator, I will still be involved when there are discussions to align visions. This involvement may reduce over time.

One potential downside of this adoption could be a change of direction of the project. After having a conversation with @tschoffelen, I don't see any strong desire to change the direction of the project in any significant way.

Let me know your thoughts!

**Comment by ceilfors** [comment]
@hugosenari Curious if you received a notification for the above and have any thoughts :)

**Reply by ceilfors** [reply]
@geoffdutton Also curious if you have any thoughts on this discussion!

**Comment by hugosenari** [comment]
I did. And have, but they aren't relevant to the subject.

Here the subjective opinionated useless thoughts:

- We have lost our most important contributor to something more important in life than code. So I'm very glad to hear that. Could be [worse](https://en.wikipedia.org/wiki/Hans_Reiser)
- Unless you can feed your child with PR, I think that it is important that open source projects to have some financial strategy to keep them running.
- While voted against't it, TypeScript is the new "Java" for corps. I still think the laconiajs is a piece of art in its architecture and simplicity, but corps don't care, they want the [enterprise solution](https://github.com/EnterpriseQualityCoding/FizzBuzzEnterpriseEdition). Means less opportunities for funding. 
- If a more enlightened company wants to keep laconiaJS going while we all have the same problems as you, we all must be happy at least or workout to give project another option.
- At least you are not begging for employment, and breaking anyone else build in the way.
- #508 AWS SDK 3. I was expecting SDK 3 to reduce the size of the build and the cold start, but AWS SDK is big because of API metadata that isn't modularized, if you use one AWS module or all, the size is almost the same (or was, I checked this 2 years ago).
- I planned to do the upgrade, but got demotivated once I realized that.
- Now isn't what we gain with, but what we wouldn't lose: **everything**.
- The motivation is different, not only for me, but for everyone. 
- Even a PR for that won't fix the project lack of contribution issue.
- As I'm not working to give project another option. Whatever you decide, I'm ok with it.

**Reply by ceilfors** [reply]
Love these thoughts, thanks Hugo.

> Unless you can feed your child with PR, I think that it is important that open source projects to have some financial strategy to keep them running.

Particularly this one!

**Comment by ceilfors** [comment]
Hey Hugo, I don't think these are useless thoughts! 

To focus specifically on another option, another option was suggested by @Ankcorn on Twitter: Get a Foundation to host it. @brianleroux suggested https://openjsf.org/ and kindly offered to help us navigate. ([The Twitter thread](https://twitter.com/brianleroux/status/1755650443525714026))

I thought it would be useful for me to document my thoughts here:
- While the ethos of the foundation (specifically OpenJSF) looks appealing, what Laconia.js needs most now is better maintenance and contributions. While a Foundation can help achieve this, they'll be doing this non-directly i.e. increase visibility -> attract more usage and contributions -> attract maintainers.
  - NearSt adoption will satisfy the needs directly.
- The upside of getting OpenJSF to host Laconia.js is its neutrality.
  - I believe NearSt will keep the project neutral, and it'll be part of the agreement in the adoption so that it won't prevent a Foundation from hosting it, if needed in the future (keeping the option open)

**Reply by hugosenari** [reply]
Get a Foundation to host it, unless this means financial support, I don't think 'hosting' (infra) is an issue now.
The only advantage is to be 'endorsed'. That may be important for corps, but not sure if it guarantees more visibility.
Means if somehow another corps discover the projects is more likely that they trust in the project, but [first they have to discover it](https://www.tweag.io/blog/2020-08-20-how-nix-grew-a-marketing-team/).

Maybe is why Apache is sometimes referred as the open source graveyard, they endorse projects but won't make them more visible than an initial "hey project X is now under our umbrella". If the project has its own momentum, ok, if not, good luck to the project.

**Reply by ceilfors** [reply]
Thanks, Hugo. I think we're aligned on those thoughts. No, I don't believe a Foundation would provide (good enough) financial support.

**Comment by ceilfors** [comment]
We have concluded for the adoption to go ahead. The list of high-level agreements in between me and NearSt can be found here:

[NearSt Adoption Summary Agreement.pdf](https://github.com/laconiajs/laconia/files/14383842/NearSt.Adoption.Summary.Agreement.pdf)

Thanks @hugosenari for your contribution to this discussion! We'll be planning some work to make the necessary transfer of the current infrastructure.

**Reply by tschoffelen** [reply]
Thanks @ceilfors! Really excited to get started actively maintaining this in addition to the existing maintainers.

Not much will change for the project as a whole, but hopefully we can give it the time and resources it deserves!
